conanfile.py

```diff
@@ -30,13 +30,22 @@ class H5cppConan(ConanFile):
     # The temporary build diirectory
     build_dir = "./%s/build" % folder_name
 
+    options = {
+        "parallel": [True, False],
+    }
     default_options = (
+        "parallel=False",
         "Boost:shared=True",
         "hdf5:shared=True",
         "gtest:shared=True"
     )
+
     generators = "cmake"
 
+    def requirements(self):
```

Comment:
Considering the comment I left in https://github.com/ess-dmsc/conan-hdf5/pull/6, I believe we should wait before we incorporate the MPICH Conan package, as the new h5cpp release will not require patching MPICH. Instead of adding the conditional MPICH requirement, we should instead use this code to set the `parallel=True` and `cxx=False` options for h5cpp to (see https://github.com/conan-io/conan/issues/1020).